pkg/app/piped/cloudprovider/lambda/function.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package lambda
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	versionV1Beta1       = "pipecd.dev/v1beta1"
+	FunctionManifestKind = "LambdaFunction"
+)
+
+type FunctionManifest struct {
+	Kind       string               `json:"kind"`
+	APIVersion string               `json:"apiVersion,omitempty"`
+	Spec       FunctionManifestSpec `json:"spec"`
+}
+
+func (fm *FunctionManifest) validate() error {
+	if fm.APIVersion != versionV1Beta1 {
+		return fmt.Errorf("unsupported version: %s", fm.APIVersion)
+	}
+	if fm.Kind != FunctionManifestKind {
+		return fmt.Errorf("invalid manifest kind given: %s", fm.Kind)
+	}
+	if err := fm.Spec.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// FunctionManifestSpec contains configuration for LambdaFunction.
+type FunctionManifestSpec struct {
+	Name     string            `json:"name"`
+	ImageURI string            `json:"image"`
+	Tags     map[string]string `json:"tags,omitempty"`
+}
+
+func (fmp FunctionManifestSpec) validate() error {
+	if len(fmp.Name) == 0 {
+		return fmt.Errorf("lambda function is missing")
+	}
+	if len(fmp.ImageURI) == 0 {
+		return fmt.Errorf("image uri is missing")
+	}
+	return nil
+}
+
+func loadFunctionManifest(path string) (FunctionManifest, error) {
```

Comment:
Could you add some tests for this function?

Reply:
sorry I forgot to commit that 🙏 actually, I wrote tests for the `parseFunctionManifest` instead 👀